avocado/core/remote/runner.py

```diff
@@ -325,8 +329,8 @@ class VMTestRunner(RemoteTestRunner):
     Test runner to run tests using libvirt domain
     """
 
-    def __init__(self, job, result_proxy):
-        super(VMTestRunner, self).__init__(job, result_proxy)
+    def __init__(self, job, result_proxy, result):
```

Comment:
As I said, because I know where are you heading, this makes sense. The problem is that this is an intermittent patch and without the future patches it makes a little sense (actually it adds another overhead, without any significant improvement. You can achieve the same level of testing by running without HumanResult, which is supported even now). For this reason I'd prefer if you bundle to the patchset, where you remove the ResultsProxy. (fixes in missing Results attr are fine now)